components/omnibox/browser/brave_bookmark_provider.cc

```diff
@@ -54,9 +54,10 @@ void BraveBookmarkProvider::Start(const AutocompleteInput& input,
     // Note: By default the BookmarkProvider will fill the URL from the
     // bookmark. This is fine upstream as they don't allow bookmarks to be the
     // default match (it breaks the user editing the input when its default).
-    // We set |fill_into_edit| to the input text so what the user typed is not
-    // modified.
+    // We set |fill_into_edit| to the input text and reset
+    // |inline_autocompletion| so what the user typed is not modified.
     match.fill_into_edit = input.text();
+    match.inline_autocompletion = u"";
```

Comment:
I'm not sure this is right fix because this change breaks inline completion behavior.
When typing `chrome://new`, `tab` is autocompleted and user can know the complete url when enters in the omnibox. I'm worring this could affect security persepective of view.
![image](https://github.com/user-attachments/assets/7b36fc84-d548-475b-bb43-658dea036ca3)


but it's not anymore with this change.
![image](https://github.com/user-attachments/assets/f3e31da1-df8d-4975-ba13-26b29cc306fd)

Reply:
Hmmm, hadn't thought of that. After a bit of investigation, I found I should probably have been using `SetAllowedToBeDefault`. This seems to fix the behavior!